gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/PolygonDrawingViewModel.java

```diff
@@ -0,0 +1,255 @@
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import timber.log.Timber;
+
+@SharedViewModel
+public class PolygonDrawingViewModel extends AbstractViewModel {
+
+  /**
+   * DISTANCE_THRESHOLD check if the first vertex of polygon is within 10 meters of distance from
+   * the current pointing location.
+   */
+  public static final int DISTANCE_THRESHOLD = 10;
+
+  @Hot private final Subject<Nil> defaultState = PublishSubject.create();
+
+  private final MutableLiveData<Integer> completeButtonVisible = new MutableLiveData<>(INVISIBLE);
+  private final MutableLiveData<Integer> addVertexButtonVisible = new MutableLiveData<>(VISIBLE);
+  /** Polyline drawn by the user but not yet saved as polygon. */
+  @Hot
+  private final MutableLiveData<PolygonFeature> drawnPolylineVertices = new MutableLiveData<>();
+
+  @Hot(replays = true)
+  private final MutableLiveData<Boolean> locationLockEnabled = new MutableLiveData<>();
+
+  private final LiveData<Integer> iconTint;
+  @Hot private final Subject<Boolean> locationLockChangeRequests = PublishSubject.create();
+  private final LocationManager locationManager;
+  private final LiveData<BooleanOrError> locationLockState;
+  private final List<Point> vertices = new ArrayList<>();
+  /** The currently selected layer and project for the polygon drawing. */
+  private final BehaviorProcessor<Optional<Layer>> selectedLayer =
+      BehaviorProcessor.createDefault(Optional.empty());
+
+  private final BehaviorProcessor<Optional<Project>> selectedProject =
+      BehaviorProcessor.createDefault(Optional.empty());
+
+  private final OfflineUuidGenerator uuidGenerator;
+  private final AuthenticationManager authManager;
+  @Nullable private Point cameraTarget;
+
+  @Inject
+  PolygonDrawingViewModel(
+      LocationManager locationManager,
+      AuthenticationManager authManager,
+      OfflineUuidGenerator uuidGenerator) {
+    this.locationManager = locationManager;
+    this.authManager = authManager;
+    this.uuidGenerator = uuidGenerator;
+    Flowable<BooleanOrError> locationLockStateFlowable = createLocationLockStateFlowable().share();
+    this.locationLockState =
+        LiveDataReactiveStreams.fromPublisher(
+            locationLockStateFlowable.startWith(BooleanOrError.falseValue()));
+    this.iconTint =
+        LiveDataReactiveStreams.fromPublisher(
+            locationLockStateFlowable
+                .map(locked -> locked.isTrue() ? R.color.colorMapBlue : R.color.colorGrey800)
+                .startWith(R.color.colorGrey800));
+  }
+
+  private Flowable<BooleanOrError> createLocationLockStateFlowable() {
+    return locationLockChangeRequests
+        .switchMapSingle(
+            enabled ->
+                enabled
+                    ? this.locationManager.enableLocationUpdates()
+                    : this.locationManager.disableLocationUpdates())
+        .toFlowable(BackpressureStrategy.LATEST);
+  }
+
+  @Hot
+  public Observable<Nil> getDefaultState() {
```

Comment:
"Default state" of what? Name is unclear without more context.

Reply:
changed to getDefaultMapMode()

Reply:
Sorry, I still don't get it.. what is the "default map mode", why is it always `Nil`, and why would the "default" change over time (`Observable`)?

Reply:
default is only used to set Mode to Default. It is set to NIL as we want to listen only when its called so that we could set the default view which is Mode.DEFAULT.